man/systemd.service.xml

```diff
@@ -897,6 +897,28 @@
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>RestartMode=</varname></term>
+
+        <listitem>
+          <para>Takes a string value that specifies how a service should restart:
+            <itemizedlist>
+              <listitem><para>If set to <option>normal</option> (the default), the service restarts by
+              going through a failed state.</para></listitem>
+
+              <listitem><para>If set to <option>quick</option>, the service restarts without
+              going through a failed state.
+              <varname>ExecStopPost=</varname> is invoked.
+              <varname>OnFailure=</varname> is skipped.</para></listitem>
+            </itemizedlist>
```

Comment:
this focusses only on the "failed" state. But your patch also disables the "inactive" state we go through between the shutting down and starting up. 

And because we enter that inactive state we also will run OnActive=. With your new option this will *also* be turned off.